src/services/housing.js

```diff
@@ -198,6 +178,48 @@ const changeApartmentAppEditor = async (applicationID, newEditorUsername) => {
   );
 };
 
+/**
+ * Helper function to fill in any missing properties of an applicant object's Profile, OffCampusProgram, etc.
+ *
+ * @function formatApplicantInfo
+ * @param {ApartmentApplicant} applicant an object representing an apartment applicant
+ * @return {ApartmentApplicant} Application details
+ */
+function formatApplicantInfo(applicant) {
+  // //! DEBUG: Temporary workaround for an API bug that causes 'Profile.PersonType' to be undefined
+  // user.getProfileInfo(applicant.Username ?? applicant.Profile.AD_Username).then((profile) => {
+  //   applicant.Profile = profile;
+  // });
+
+  applicant.Profile.PersonType = 'stu';
+  user.setFullname(applicant.Profile);
+  user.setClass(applicant.Profile);
+
+  // The following 'Class' property is needed for the staff page
+  if (applicant.Class === null || Number(applicant.Class)) {
+    // Use converted Class from number ('1', '2', '3', ...) to words ('Freshman', 'Sophomore', ...)
+    applicant.Class = applicant.Profile.Class;
+  }
+
+  applicant.OffCampusProgram ?? (applicant.OffCampusProgram = '');
```

Comment:
This is the intent. However, I made an issue about a babel compile error, there may be a dependency issue or a module config issue preventing the `??=` from compiling

I used the `a ?? (a = b)` as a workaround because the documentation for `a ??= b` explained that it is functionally equivalent

Reply:
I described the issue in #1101